crypto/evp/evp_lib.c

```diff
@@ -296,6 +360,31 @@ int EVP_CIPHER_CTX_nid(const EVP_CIPHER_CTX *ctx)
     return ctx->cipher->nid;
 }
 
+int EVP_CIPHER_mode(const EVP_CIPHER *cipher)
+{
+    if (cipher->prov != NULL) {
+        int mode;
+
+        /* Cipher comes from a provider - so ask the provider for the mode */
+        const OSSL_PARAM params[] = {
+            OSSL_PARAM_int(OSSL_CIPHER_PARAM_MODE, &mode),
```

Comment:
I think you need to use this instead.
OSSL_PARAM params[2];
size_t n = 0;

params[n++]  = OSSL_PARAM_construct_int(..);
params[n++] = OSSL_PARAM_construct_end();

Reply:
This was fixed in #8700